.github/workflows/terraform_gh_runner.yml

```diff
@@ -1,16 +1,18 @@
-name: Terraform Workflow
+name: Terraform GH Runner Deployment
 
 on:
-  push:
+  workflow_dispatch:
+  pull_request:
     branches:
       - main
     paths:
-      - './github-runners/terraform/base/**'
-  workflow_dispatch:
+      - "./github-runners/terraform/base/**"
 
 jobs:
   terraform_gh_runner:
     runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It doesn't seem to be used, why setting?

Reply:
The env variable needs to be set when calling github api, otherwise an error is thrown.

Reply:
What and where calls those APIs? I don't see any usage of it.

Reply:
Ok, for $token in the auth header of the curl request that is the pat token to authenticate to the repo and get the runner token. If i remove GH_TOKEN you will see an error in ci that you need to have GH_TOKEN to call github apis from workflow :). I wasn't using it before, but added it to fix the problem. Here is an old version:
https://github.com/subspace/infra/blob/bd296590f29cdd730ab29869c7ade1258cb395ae/.github/workflows/terraform_gh_runner.yml

Reply:
Then show me where it is used :man_shrugging: You have linked to `curl` request that doesn't use this environment variable as a reason for this environment variable to exist. It makes no sense to me.

Reply:
@nazar-pc https://github.com/subspace/infra/actions/runs/6484473796/job/17608384650#step:5:24

Reply:
Okay, so it is not for `curl`, it was for `gh` CLI command, now it makes sense.

With that knowledge we understand why it is necessary. I think what we should do now in a secure way without using heavy tools like setting repo (I'm fairly certain setting environment variable wasn't, check CI logs to invalidate any tokens that were printed if you have tested it before) without using heavy tools like setting repo's secrets from workflow is to use outputs and mask them such that they are not visible in logs: https://github.com/orgs/community/discussions/25225#discussioncomment-3246942

That is the goal here: to pass the token from one step into another. What we had in earlier versions of this PR are various suboptimal/incorrect ways of achieving that ultimate goal.

Reply:
correct, github cli, I should have been more clear about that "The GITHUB_TOKEN is needed in CI or bash cli for API calls" but couldn't remember exactly since it was a while ago. That being said the solution of masking the secret and just passing it is nice if using the ephemeral runners maybe but this solution and workflow is for the dedicated runners, where i need to retain the secret so I can unregister and delete runner if need be and remove it from github. See https://docs.github.com/en/free-pro-team@latest/rest/actions/self-hosted-runners?apiVersion=2022-11-28#delete-a-self-hosted-runner-from-an-organization

Reply:
But you mentioned that tokens only live for 1 hour, so you'll have to retrieve fresh token anyway. Why retaining it then at all?

Reply:
The token is not needed to deregister, a force delete can be done. So i've used the masking technique.